extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1984,8 +2028,12 @@ private boolean isTaskCurrent(int taskGroupId, String taskId)
 
     final List<ListenableFuture<Void>> futures = Lists.newArrayList();
     for (Map.Entry<String, TaskData> entry : taskGroup.tasks.entrySet()) {
-      if (!entry.getValue().status.isComplete()) {
-        futures.add(stopTask(entry.getKey(), false));
+      final String taskId = entry.getKey();
+      final TaskData taskData = entry.getValue();
+      if (taskData.status == null) {
+        killTask(taskId);
```

Comment:
Is killing the task the right idea here? If we don't know its status is it safe to leave it alone?

Reply:
The null `taskData.status` means that the supervisor hasn't updated it yet, so its actual status can be anything. I think this should kill tasks if their status are unknown because this method is supposed to stop all tasks in the given taskGroup.

Reply:
Got it, that sounds good.